src/api/app/models/token/workflow.rb

```diff
@@ -17,7 +17,12 @@ def call(options)
     yaml_file = Workflows::YAMLDownloader.new(@scm_webhook.payload, token: self).call
     @workflows = Workflows::YAMLToWorkflowsService.new(yaml_file: yaml_file, scm_webhook: @scm_webhook, token: self, workflow_run_id: options[:workflow_run].id).call
 
-    @workflows.each(&:call) if validation_errors.none?
+    return validation_errors unless validation_errors.none?
+
+    # This is just an initial generic report to give a feedback asap. Initial status pending
+    ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token).call
+    @workflows.each(&:call)
+    ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token, 'success').call
```

Comment:
So what happens if all the workflows fails? The initial status will be forever in yellow right?

Can we handle this?

Reply:
Yes, that was my wish too.. that for example, if any exception happens while running the workflows, it could be reflected as well in that message. I still have to figure out how to make it.. i was thinking in something like:

```
# body of the method call
workflow_status = 'success'
ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token).call
@workflows.each(&:call)
rescue StandardError
 workflow_state = 'failure'
ensure 
  ScmInitialStatusReporter.new(@scm_webhook.payload, @scm_webhook.payload, scm_token, workflow_state).call
  validation_errors 
end
```

Reply:
Okay, doesn't have to be here. We can push this out if you want to.

Reply:
yes, I spent sometime golfing around with this code and I'm still not happy with the solution using `ensure`

Reply:
please do not forget to create a card in the backlog

Reply:
https://trello.com/c/HDjqdG9t/2501-make-the-initial-status-report-more-robust